### PR TITLE
:customcss: attribute for easy per-presentation CSS

### PR DESCRIPTION
**Pull request by obilodeau**
Check the documentation commit for details.

### CODE REVIEW

**Comment by mojavelinux** [comment]
I like the capability, but we're changing the meaning of the linkcss attribute attribute here. Therefore, I think we should use a different attribute name. We could align with the deck.js converter and use the customcss attribute instead (or perhaps revealjs_customcss). If empty, it resolved to asciidoctor-reveal.css. If non-empty, the value provided is used.

**Comment by obilodeau** [comment]
Sounds fair. I'll go with `customcss` since this is not related to an upstream feature.

**Comment by mojavelinux** [comment]
:+1:

**Comment by cmoulliard** [comment]
Don't forget to also update the jade template !

**Comment by obilodeau** [comment]
@cmoulliard as I said in #63, I won't do the jade templates. I love asciidoc[tor] and I love sharing about infosec with nice HTML presentations but I don't have enough time to learn properly ruby and javascript plus slim and jade/pug. I already feel my contributions are not [idiomatic](http://mrjoelkemp.com/2013/05/what-is-idiomatic-programming/) to a ruby programmer (being a Python/Perl "programmer") and I don't want to double that mess. Someone needs to step in to do the jade thing or we need to switch to a ruby/javascript accessible template engine.

**Comment by obilodeau** [comment]
@mojavelinux I implemented what we discussed and added a test case. From my perspective, this is ready to go.

File: templates/slim/document.html.slim [hunk 1]
```diff
@@ -65,6 +65,10 @@ html lang=(attr :lang, 'en' unless attr? :nolang)
       document.write( '<link rel="stylesheet" href="#{revealjsdir}/css/print/' + ( window.location.search.match( /print-pdf/gi ) ? 'pdf' : 'paper' ) + '.css" type="text/css" media="print">' );
     - unless (docinfo_content = docinfo :header, '.html').empty?
       =docinfo_content
+    - if attr? :customcss and (attr :customcss).empty?
```
Comment (mojavelinux) [inline templates/slim/document.html.slim]: Here's my preferred way of writing this:

```slim
- if attr? :customcss
  link rel='stylesheet' href=((customcss = attr :customcss).empty? ? 'asciidoctor-revealjs.css' : customcss)
```

I'd also be open to:

```slim
- if attr? :customcss
  - if (customcss = attr :customcss).empty?
    link rel='stylesheet' href='asciidoctor-revealjs.css'
  - else
    link rel='stylesheet' href=customcss
```

**Comment by mojavelinux** [comment]
> I already feel my contributions are not idiomatic to a ruby programmer (being a Python/Perl "programmer")

I think you're doing a fine job. We're learning together. Besides, we use a unique Ruby style in this project anyway, and I kind of like that.

**Comment by cmoulliard** [comment]
Even if you can't maintain the templates in jade, it is important to list somewhere the changes applied to slim and/or jade.

File: templates/slim/document.html.slim (hunk 1 above)
Comment (obilodeau) [inline templates/slim/document.html.slim]: I noticed that you love the ternary operator :)

Comment (obilodeau) [inline templates/slim/document.html.slim]: The test/assignment is very interesting, I'll try to remember that trick!

Comment (mojavelinux) [inline templates/slim/document.html.slim]: I love it to death. It helps cut down on a lot of lines of code and useless assignments. Combined with the assignment in the conditional, I can express meaning much more succinctly.

**Pull request merged**